src/main/java/com/netgrif/application/engine/orgstructure/web/GroupController.java

```diff
@@ -50,6 +52,47 @@ public GroupsResource getAllGroups() {
         return new GroupsResource(groupResponse);
     }
 
+    @Authorizations(value = {
+            @Authorize(authority = "GROUP_VIEW_MY")
+    })
+    @ApiOperation(value = "Get logged user's groups in the system",
+            notes = "Caller must have the GROUP_VIEW_MY authority",
+            authorizations = @Authorization("BasicAuth"))
+    @RequestMapping(value = "/my", method = RequestMethod.GET, produces = MediaTypes.HAL_JSON_VALUE)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = GroupsResource.class),
+            @ApiResponse(code = 403, message = "Caller doesn't fulfill the authorisation requirements"),
+    })
+    public GroupsResource getMyGroups(Authentication auth) {
+        String loggedUserId = ((LoggedUser)auth.getPrincipal()).getId();
+        List<Case> groups = service.findAllGroups();
+        Set<Group> groupResponse = groups.stream()
+                .filter(aCase -> aCase.getAuthor().getId().equals(loggedUserId))
```

Comment:
1. logic should not be implemented in a controller class
2. filtering should be done in the database

Reply:
Resolved.